bin/check_samplesheet.py

```diff
@@ -1,58 +1,69 @@
-# GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License
-# along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-####################################################################################################
+#!/usr/bin/env python
 
+import os
 import sys
-import os.path
-import csv
-import pandas as pd
-import io
+import errno
 import argparse
+import pandas as pd
 
 
 def parse_args(args=None):
-    parser = argparse.ArgumentParser()
-    parser.add_argument('-i', "--input", required=True, metavar='FILE', type=argparse.FileType('r'), help="Input file containing: condition, type, microbiome_path, alleles, weights_path.")
+    Description = "Reformat nf-core/metapep samplesheet file and check its contents."
+    Epilog = "Example usage: python check_samplesheet.py <FILE_IN>"
+
+    parser = argparse.ArgumentParser(description=Description, epilog=Epilog)
+    parser.add_argument('-i', "--input", required=True, metavar='FILE', type=argparse.FileType('r'), help="Input samplesheet file containing: condition, type, microbiome_path, alleles, weights_path.")
     parser.add_argument('-m', "--microbiomes", required=True, metavar='FILE', type=argparse.FileType('w'), help="Output file containing: microbiome_id, microbiome_path, microbiome_type, weights_path.")
     parser.add_argument('-c', "--conditions", required=True, metavar='FILE', type=argparse.FileType('w'), help="Output file containing: condition_id, condition_name, microbiome_id.")
     parser.add_argument('-a', "--alleles", required=True, metavar='FILE', type=argparse.FileType('w'), help="Output file containing: allele_id, allele_name.")
     parser.add_argument('-ca', "--conditions_alleles", required=True, metavar='FILE', type=argparse.FileType('w'), help="Output file containing: condition_id, allele_id.")
     return parser.parse_args(args)
 
 
-def main(args=None):
-    args = parse_args(args)
+def make_dir(path):
+    if len(path) > 0:
+        try:
+            os.makedirs(path)
+        except OSError as exception:
+            if exception.errno != errno.EEXIST:
+                raise exception
+
+
+def print_error(error, context="Line", context_str=""):
+    error_str = "ERROR: Please check samplesheet -> {}".format(error)
+    if context != "" and context_str != "":
+        error_str = "ERROR: Please check samplesheet -> {}\n{}: '{}'".format(
+            error, context.strip(), context_str.strip()
+        )
+    print(error_str)
+    sys.exit(1)
 
-    input_table = pd.read_csv(args.input, sep='\t')
+# TODO for 'proteins' type
+# - allow weight input for type 'proteins' as well! (for now use equal weight ?)
 
-   # check if microbiome_path file extensions are valid
+def check_samplesheet(args):
+    """
+    This function checks that the samplesheet follows the following structure:
+
+    condition,type,microbiome_path,alleles,weights_path
+
+    For an example see:
+    TODO
```

Comment:
you can lint to the sample sheet in folder `assets`?

Reply:
I linked to a sample sheet in nf-core/test-datasets, is that ok? Do I need a sample sheet in `assets`?

Reply:
Usually, there is a samplesheet in assets for documentation purposes only. The sample sheets in nf-core/test-datasets are for test purposes only. So I'd say, yes, there should be a sample sheet in assets.